swaggerhub-wso2-integration/src/main/java/com/swagger/plugins/wso2/MakeHttpRequestService.java

```diff
@@ -0,0 +1,126 @@
+import org.apache.http.impl.client.HttpClients;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ *
+ * Class with the implementations of the methods defined in HttpRequestService interface
+ *
+ */
+public class MakeHttpRequestService implements HttpRequestService {
+
+    private static final Logger log = LoggerFactory.getLogger(MakeHttpRequestService.class);
+
+    /**
+     * Makes a POST request to the given url and returns the response
+     *
+     * @param url                       URL that the http call should be made to
+     * @param tokenPrefix               Prefix of the token of the request
+     * @param token                     Authorization token of the request
+     * @param contentType               Content type of the request
+     * @param payload                   Body of the request
+     * @return                          Returns a Http response
+     * @throws PluginExecutionException Thrown when an exception is caught while the plugin executes
+     */
+    public HttpResponse makePostRequest(String url, String tokenPrefix, String token, String contentType,
+                                        StringEntity payload) throws PluginExecutionException {
+        HttpResponse response;
+        HttpClient httpClient = HttpClients.createDefault();
+        HttpPost postRequest = new HttpPost(url);
+        postRequest.setHeader("Authorization", tokenPrefix + " " + token);
+        postRequest.setHeader("Content-Type", contentType);
+        postRequest.setEntity(payload);
+        try {
+            response = httpClient.execute(postRequest);
+        } catch (IOException e) {
+            log.error("Error making POST request", e);
+            throw new com.smartbear.swaggerhub.plugins.
+                    PluginExecutionException(PluginExecutionException.INVALID_INPUT, "Error making POST request");
+        }
+        return response;
+    }
+
+    /**
+     * Makes a PUT request and returns the response
+     *
+     * @param url                       URL that the http call should be made to
+     * @param tokenPrefix               Prefix of the token of the request
+     * @param token                     Authorization token of the request
+     * @param contentType               Content type of the request
+     * @param payload                   Body of the request
+     * @throws PluginExecutionException Thrown when an exception is caught while the plugin executes
+     */
+    public HttpResponse makePutRequest(String url, String tokenPrefix, String token, String contentType,
+                                       StringEntity payload) throws PluginExecutionException {
+        HttpResponse response;
+        HttpClient httpClient = HttpClients.createDefault();
+        HttpPut putRequest = new HttpPut(url);
+        putRequest.setHeader("Authorization", tokenPrefix + " " + token);
+        putRequest.setHeader("Content-Type", contentType);
```

Comment:
Shall we move Authorization and Content-Type into constants?

Reply:
Moved